github/packages/github/github_list.js

```diff
@@ -0,0 +1,248 @@
+ * @param {!object} [secrets = {}] list of secrets
+ * @return {Promise<SlackBodyType>} Response body
+ */
+const axios = require('axios');
+
+const requestThreshold = 3;
+const headers = {
+  'Content-Type': 'application/json',
+};
+async function getRequest(url, secrets) {
+  if (secrets.github_token) headers.Authorization = `Bearer ${secrets.github_token}`;
+  return (axios({
+    method: 'get',
+    url,
+    headers,
+  }).then((res) => res).catch((err) => err));
+}
+
+async function command(params, commandText, secrets = {}) {
+  let {
+    entity, // repositories, commits, code, issues pull requests, users, topics
+    keywords,
+    repository,
+    language,
+  } = params;
+  let displayEntity = entity;
+  const displayKeywords = keywords;
+  let pageSize = '';
+  switch (entity) {
+    case 'r':
+    case 'rep':
+    case 'repo':
+    case 'repos':
+      displayEntity = 'Repositories';
+      entity = 'repositories';
+      break;
+    case 'cm':
+    case 'commit':
+      entity = 'commits';
+      displayEntity = 'Commits';
+      headers.Accept = 'application/vnd.github.cloak-preview';
+      pageSize = '&per_page=20';
+      break;
+    case 'c':
+    case 'cd':
+      entity = 'code';
+      displayEntity = 'Code Files';
+      keywords += '+in:file';
+      if (!repository) return fail('*please specify a repository, using -r flag e.g.*\n`/nc github_list c github -r nimbella/command-sets`');
+      break;
+    case 'i':
+    case 'issue':
+      entity = 'issues';
+      displayEntity = 'Issues';
+      keywords += '+is:issue';
+      pageSize = '&per_page=5';
+      break;
+    case 'p':
+    case 'pr':
+    case 'prs':
+      entity = 'issues';
+      displayEntity = 'Pull-requests';
+      keywords += '+is:pr';
+      pageSize = '&per_page=20';
+      break;
+    case 'u':
+    case 'user':
+      displayEntity = 'Users';
+      entity = 'users';
+      break;
+    case 't':
+    case 'topic':
+      entity = 'topics';
+      displayEntity = 'Topics';
+      headers.Accept = 'application/vnd.github.mercy-preview+json';
+      pageSize = '&per_page=10';
+      break;
+    default:
+      displayEntity = 'Repositories';
+      entity = 'repositories';
+      break;
+  }
+  const url = `https://api.github.com/search/${entity}?q=${repository ? `repo:${repository}+` : ''}${keywords}${language ? `+language:${language}` : ''}${pageSize}`;
+  const res = await getRequest(url, secrets);
+
+  if (res && res.data) {
+    const tokenMessage = secrets.github_token ? '' : '*For greater limits you can add <https://help.github.com/en/github/authenticating-to-github/creating-a-personal-access-token-for-the-command-line | secrets> using*\n `/nc secret_create`';
+    const currReading = parseInt(res.headers['x-ratelimit-remaining']);
+    let header = `*${displayEntity}* with keywords _*${displayKeywords}*_`;
+    if (currReading < requestThreshold) {
+      header = `:warning: *You are about to reach the api rate limit.* ${tokenMessage}`;

```

Comment:
Should the string be appended?

Reply:
the header message is ceremonial, I think it should give way for a more important message.

Reply:
Okay. I just wanted to make sure you didn't intend to append.